### PR TITLE
feat: initial testing suite setup

### PR DESCRIPTION
**Pull request by jonaslagoni**
This PR adds the initial testing suite setup which includes:
- A JSON Schema which validates the test suite files.
- A GitHub CI workflow that validates the schema against all added testing suite files on each PR.
- Two basic testing suite files:
  - `name` uses the new keyword for defining the name of the generated entity (class in this instance).
  - `properties` shows a basic example of expected behavior for the types `string`, `boolean`, `number`, `null`.

Nothing is final when merging this PR, it is a WIP and provides a baseline for further development. `name` might not be the right keyword to introduce, which we can discuss later.

After this PR I will provide a new one with a basic TS runner so we can validate the work as we go. This way we make sure that the testing suite is actually useable from start to finish.

### CODE REVIEW

**Comment by jonaslagoni** [comment]
You can see the action being run here: https://github.com/jonaslagoni/vocab-idl/runs/3827648402?check_suite_focus=true

**Comment by karenetheridge** [comment]
Are you familiar with https://github.com/json-schema-org/JSON-Schema-Test-Suite? It defines a format for testing schemas, and there's been some amount of tooling built around that for interpreting those files and sending the files through various evaluator implementations.

File: test-suite/schema.json [hunk 1]
```diff
@@ -0,0 +1,143 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "definitions": {
```
Comment (karenetheridge) [inline test-suite/schema.json:3]: In draft2020-12, this is `$defs`.

**Comment by karenetheridge** [comment]
Yes, I guess I don't understand what the test suite is describing, if not true/false validation results. How would I use these files to verify my implementation of the vocabulary?

File: test-suite/schema.json (hunk 1 above)
Comment (jonaslagoni) [reply test-suite/schema.json:3]: Ahh, right!

**Comment by jonaslagoni** [comment]
> Yes, I guess I don't understand what the test suite is describing, if not true/false validation results. How would I use these files to verify my implementation of the vocabulary?

@karenetheridge we are not verifying that a schema validates correctly against data but we are verifying that the schema accurately gets generated using reflection🙂This is based on @jdesrosiers example from [our discussion](https://github.com/json-schema-org/vocab-idl/discussions/8#discussioncomment-1229259).

File: test-suite/schema.json [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        "tests": {
+          "type": "array",
+          "prefixItems": [
+            {
+              "$ref": "#/$defs/hasNameAssertion"
+            },
+            {
+              "$ref": "#/$defs/hasTypeAssertion"
+            }
+          ]
```
Comment (jdesrosiers) [inline test-suite/schema.json:97]: Did you really mean to use `prefixItems` here?

```suggestion
          "items": {
            "anyOf": [
              {
                "$ref": "#/$defs/hasNameAssertion"
              },
              {
                "$ref": "#/$defs/hasTypeAssertion"
              }
            ]
          }
```

File: test-suite/idl-schema/properties/properties.json [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+          },
+          {
+            "assertion": "hasProperty",
+            "tests": [
+              {
+                "assertion": "hasName",
+                "arguments": [
+                  "StringProperty"
+                ]
+              },
```
Comment (jdesrosiers) [inline test-suite/idl-schema/properties/properties.json:43]: This should be an argument to the "hasProperty" assertion, not an assertion itself. Once you've determined that there exists a property with this name, then you can do additional tests on the property that was found ("tests") such as asserting it's type.

File: test-suite/schema.json (hunk 2 above)
Comment (jonaslagoni) [reply test-suite/schema.json]: That's a good point, changing 🙂

File: test-suite/idl-schema/properties/properties.json (hunk 3 above)
Comment (jdesrosiers) [reply test-suite/idl-schema/properties/properties.json:43]: Right. Most assertions should have "arguments". We can't check if a type has a property if we don't tell it the name of the property check for.

Comment (jonaslagoni) [reply test-suite/idl-schema/properties/properties.json:43]: Yea, good point, changing.

Comment (jdesrosiers) [reply test-suite/idl-schema/properties/properties.json]: The same thing applies to "hasClass". That needs to be updated as well.

Comment (jonaslagoni) [reply test-suite/idl-schema/properties/properties.json]: Good catch, changed 🙂

**Review by jdesrosiers — APPROVED** [review]

**Pull request merged**